### PR TITLE
Pull out roadmap into a function; add alttext to status

### PR DESCRIPTION
**Pull request by slavfox**
TL;DR - **Even more accessibility improvements.**

The last PR got me thinking - there are people using screenreaders or
non-graphical browsers, and with the current approach, they wouldn't be
able to know the status of roadmap elements. This commit does two
things:

* First, it pulls out the roadmap data into YAML files, so that editing
  it is at least as easy, and at most even easier, than editing the raw
  HTML and using CSS class for status - and adds a recursive function
  that handles the logic for displaying the roadmap (a neat Jekyll trick
  I learned while writing Github Pages docs for Foxlang). That way, we
  can easily show the roadmap elsewhere or add a new roadmap with
  minimal effort, and roadmap.html is a bit easier to read in an editor.

* Second, it removes the `::before` attribute from the list elements,
  and instead uses an actual `` tag to display the status icons -
  and assigns `alt` attributes. This way, if someone's using a
  non-graphical browser or a screen reader, they will still see a
  "Status: todo" etc. text, while users with full image support will see
  no change from the current state.

### CODE REVIEW

**Comment by PaulBone** [comment]
Hi @slavfox.

I've written a CONTRIBUTING.md file for this repository.  To make things simple and very very black-and-white. Would you mind adding a comment to this issue saying that you retroactively agree that your two contributions are CC BY-SA 4.0.

Thanks.

**Comment by slavfox** [comment]
I absolutely do agree that my contributions are CC BY-SA 4.0. However, to make it even easier, I also claim no copyright for them, and instead pass it to you, in case you decide to change the license later on.

Will this suffice? :)

**Comment by PaulBone** [comment]
Cool.  I'm not sure how "no claim" works legally and in each of our countries.  But I'll update the AUTHORS file to show this in the morning.  Either way this statement is 100% cool.

Thanks again.

**Comment by slavfox** [comment]
I can mail you a signed contract saying that if I ever decide to claim copyright I shall be beheaded and burned at the stake, or something ;)

I believe copyright transfer can only be done through a specific signed contract, in which case a supplementary statement is in order:

Regardless of whether my rescidnment of copyright is not considered legally binding, I also expressly agree for you to alter or change the license under which mt contributions have been made.

There, that should do it!

**Comment by PaulBone** [comment]
Yeah, that's cool too.

Yeah, AFAIK to actually transfer copyright you need a contract and some thing in exchange (some organisations use a $1 or a chocolate bar).  I'm not interested in asking people to jump through those hoops, or to likewise sign a contributor agreement (which is kind-of what your last message did). I am interested in protecting any contributors from me (it makes it very clear that I cannot change the license without your agreement, which you just gave) and me from you (protects me from you saying later on that your code is actually commercial and I can't use/modify it).  I suppose I should add something where you agree to indemnify the rest of the team in the case that you later discover that your code cannot be distributed under those terms.  Like if your employer turns around sand says "No, that's ours".

So yeah, I can't imagine changing the licenses and I don't want people to avoid contributing because there's a scary contributor agreement (not that they're necessarily bad, but some people might see them that way).

Cheers.
